rpcs3/Emu/Cell/PPUTranslator.cpp

```diff
@@ -4246,7 +4246,11 @@ void PPUTranslator::FCTIW(ppu_opcode_t op)
 	const auto xormask = m_ir->CreateSExt(m_ir->CreateFCmpOGE(b, ConstantFP::get(GetType<f64>(), std::exp2l(31.))), GetType<s32>());
 
 	// fix result saturation (0x80000000 -> 0x7fffffff)
+#ifdef ARCH_X64
 	SetFpr(op.frd, m_ir->CreateXor(xormask, Call(GetType<s32>(), "llvm.x86.sse2.cvtsd2si", m_ir->CreateInsertElement(GetUndef<f64[2]>(), b, u64{0}))));
+#else
+	SetFpr(op.frd, m_ir->CreateXor(xormask, Call(GetType<s32>(), "llvm.aarch64.neon.fcvtns.i32.f64", b)));
+#endif
```

Comment:
I might be dumb, but why are you calling the `x86` instruction inside `ifdef ARCH_X64`, and the `aarch64` inside the `else`. Shouldn't it be the opposite way?

Reply:
X64 is not aarch6

Reply:
It basically says "if ARCH_x64 (aka Intel/amd64), use SSE2; else, assume it's AArch64 (aka ARM64) and use Neon."